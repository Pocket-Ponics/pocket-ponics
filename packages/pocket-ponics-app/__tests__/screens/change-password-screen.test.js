import React from 'react'
import { shallow } from 'enzyme'
import APIUtil from '../../util/api-util'
jest.mock('../../util/api-util')
import { TextInput, TouchableOpacity, Alert, AsyncStorage, Platform } from 'react-native'

import ChangePasswordScreen from '../../screens/change-password-screen'

jest.mock('../../util/auth-util')
jest.mock('expo-permissions')

describe('Change Password Screen', () => {
	test('renders correctly', () => {
		const tree = shallow(
			<ChangePasswordScreen />
		).html()

		expect(tree).toMatchSnapshot()
	})

	test('changes password fields', () => {
		Platform.OS = 'ios'
		const tree = shallow(<ChangePasswordScreen />)
		Platform.OS = 'mock-platform'

		tree.find(TextInput).at(0).props().onChangeText('old password')
		tree.find(TextInput).at(1).props().onChangeText('new password')
		tree.find(TextInput).at(2).props().onChangeText('new password verification')

		expect(tree.html()).toMatchSnapshot()
	})

	test('fails to submit with password < 8 char', () => {
		jest.spyOn(Alert, 'alert').mockReturnValue(true)

		const tree = shallow(<ChangePasswordScreen />)

		tree.find(TextInput).at(1).props().onChangeText('short')
		tree.find(TouchableOpacity).at(0).props().onPress()

		expect(Alert.alert).toHaveBeenCalled()
		expect(APIUtil.changePassword).not.toHaveBeenCalled()
	})

	test('fails to submit with passwords not matching', () => {
		jest.spyOn(Alert, 'alert').mockReturnValue(true)

		const tree = shallow(<ChangePasswordScreen />)

		tree.find(TextInput).at(1).props().onChangeText('new password')
		tree.find(TextInput).at(2).props().onChangeText('new password verification')
		tree.find(TouchableOpacity).at(0).props().onPress()

		expect(Alert.alert).toHaveBeenCalled()
		expect(APIUtil.changePassword).not.toHaveBeenCalled()
	})

	test('fails to submit with server error', async () => {
		jest.spyOn(Alert, 'alert').mockReturnValue(true)
		APIUtil.changePassword.mockResolvedValueOnce({
			201: 'Invalid Password'
		})

		const navigation = {
			navigate: jest.fn(),
			getParam: () => 'mockUserName'
		}

		const tree = shallow(<ChangePasswordScreen navigation={navigation}/>)

		tree.find(TextInput).at(1).props().onChangeText('new password')
		tree.find(TextInput).at(2).props().onChangeText('new password')
		await tree.find(TouchableOpacity).at(0).props().onPress()

		expect(APIUtil.changePassword).toHaveBeenCalled()
		expect(navigation.navigate).not.toHaveBeenCalled()
	})

	test('sucessfully changes password', async () => {
		jest.spyOn(Alert, 'alert').mockReturnValue(true)
		jest.spyOn(AsyncStorage, 'setItem').mockResolvedValue(true)

		APIUtil.changePassword.mockResolvedValueOnce({
			200: 'Sucessfully saved password'
		})

		const navigation = {
			navigate: jest.fn(),
			getParam: () => 'mockUserName'
		}

		const tree = shallow(<ChangePasswordScreen navigation={navigation}/>)

		tree.find(TextInput).at(1).props().onChangeText('new password')
		tree.find(TextInput).at(2).props().onChangeText('new password')
		await tree.find(TouchableOpacity).at(0).props().onPress()

		expect(APIUtil.changePassword).toHaveBeenCalled()
		expect(AsyncStorage.setItem).toHaveBeenCalledTimes(2)
	})
})