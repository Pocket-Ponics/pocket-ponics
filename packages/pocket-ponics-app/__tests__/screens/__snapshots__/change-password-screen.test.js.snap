// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Password Screen changes password fields 1`] = `"<View style=\\"flex:1;background-color:#472600\\"><View style=\\"flex:1;align-items:center;justify-content:center;padding:20px;margin-top:40px;width:100%\\"><Image source=\\"1\\" style=\\"height:200px;width:200px;resize-mode:contain\\"></Image><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Old Password\\" placeholderTextColor=\\"rgba(255, 255, 255, 0.7)\\" value=\\"old password\\" autoCapitalize=\\"none\\" textContentType=\\"password\\"></TextInput></View><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"New Password\\" placeholderTextColor=\\"rgba(255, 255, 255, 0.7)\\" value=\\"new password\\" autoCapitalize=\\"none\\" textContentType=\\"newPassword\\"></TextInput></View><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Verify New Password\\" placeholderTextColor=\\"rgba(255, 255, 255, 0.7)\\" value=\\"new password verification\\" autoCapitalize=\\"none\\" textContentType=\\"newPassword\\"></TextInput></View><View style=\\"opacity:1\\"><Text style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;width:100%;align-items:center\\">Change Password</Text></View></View></View>"`;

exports[`Change Password Screen renders correctly 1`] = `"<View style=\\"0:[object Object];1:[object Object]\\"><View style=\\"flex:1;align-items:center;justify-content:center;padding:20px;margin-top:40px;width:100%\\"><Image source=\\"1\\" style=\\"height:200px;width:200px;resize-mode:contain\\"></Image><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Old Password\\" placeholderTextColor=\\"rgba(255, 255, 255, 0.7)\\" value=\\"\\" autoCapitalize=\\"none\\" textContentType=\\"password\\"></TextInput></View><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"New Password\\" placeholderTextColor=\\"rgba(255, 255, 255, 0.7)\\" value=\\"\\" autoCapitalize=\\"none\\" textContentType=\\"newPassword\\"></TextInput></View><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Verify New Password\\" placeholderTextColor=\\"rgba(255, 255, 255, 0.7)\\" value=\\"\\" autoCapitalize=\\"none\\" textContentType=\\"newPassword\\"></TextInput></View><View style=\\"opacity:0.3\\"><Text style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;width:100%;align-items:center\\">Change Password</Text></View></View></View>"`;
